### PR TITLE
Fixed a bug where NiftyPerforce would cause p4v to litter the local workspace folders with log files.

### PR DESCRIPTION
**Pull request by gamma09**
I noticed that whenever NiftyPerforce runs a perforce command on a file (such as view in p4v), it would end up polluting the workspace directory where the targeted file lives with a debug.log file containing this message:

`[0518/111147.930:WARNING:resource_bundle_qt.cpp(115)] locale_file_path.empty() for locale`

It appears the perforce executables (or at least p4v.exe) expect the working directory to be the perforce install directory, or else it can't find the Qt resource packs.

### CODE REVIEW

**Comment by belkiss** [comment]
Oh nice! I have noticed those files but never took the time to investigate!

I wonder if we could just change the directory when invoking p4v?

Thanks!

**Comment by gamma09** [comment]
> I wonder if we could just change the directory when invoking p4v?

Done! To be safe, though, I left p4vc using the perforce install directory because it is a wrapper around p4v for some operations.

File: NiftyPerforce/P4Operations.cs [hunk 1]
```diff
@@ -233,13 +234,13 @@ public static bool DiffFile(string filename)
 
             // Let's figure out if the user has some custom diff tool installed. Then we just send whatever we have without any fancy options.
             if (g_p4customdiff)
-                return AsyncProcess.Schedule("p4.exe", GetUserInfoString() + " diff \"" + EscapeP4Path(filename) + "#have\"", dirname, new AsyncProcess.OnDone(UnlockOp), token);
+                return AsyncProcess.Schedule("p4.exe", GetUserInfoString() + " diff \"" + EscapeP4Path(filename) + "#have\"", Path.GetDirectoryName(filename), new AsyncProcess.OnDone(UnlockOp), token);
```
Comment (belkiss) [inline NiftyPerforce/P4Operations.cs:237]: use dirname here, since it's set line 233, and reuse it line 243 for the diff -du invocation

**Comment by belkiss** [comment]
Merged! Thank you again very much! I'll tag a new version asap

**Comment by belkiss** [comment]
Seems none of the p4vc commands work anymore after this MR if the global environment variables are not set :(

I'll probably have to backout this change unless we can find a way to indicate where to find the resources

**Comment by gamma09** [comment]
> Seems none of the p4vc commands work anymore after this MR if the global environment variables are not set :(

Which global environment variables are you talking about?

**Comment by belkiss** [comment]
I mean P4CLIENT, P4PORT, P4USER, either as regular env var or set with p4 set

**Pull request merged**